### PR TITLE
[Issue Refund] Confirm and submit refund

### LINKED ISSUE #2769: [Issue Refunds] Confirm Refunds
After tapping next on the refund summary view, a confirmation alert should be presented before doing the actual refund and dismissing the navigation flow.

confirm

## Goals

- Show the dialog. 
- Submit refund to the API
- Show a notice if successful or not. 
- Navigate back to the order details if successful. 
- Present a useful error to the user if possible.
- The order details should show the submitted refund.
- Support automatic refunds (e.g. Stripe)

### LINKED ISSUE #3033: [Issue Refund] Support api_refund property
API exposes an `api_refund` property that lets the store to automatically refund payments using the order payment gateway.

We need to:
- See what Android is doing with that flag
- Research if we can fetch that value that from the order payment gateway
- Mark that flag if the payment gateway supports it

### PR DESCRIPTION
**Pull request by Ecarrion**
closes #3033 
closes #2769 

# Why

This PR connects the wires and gives the ability to a user to issue a refund! 🎉 

_PS: Handling of previous shipping refunds is not yet supported._

# How
The above is accomplished by:
**Show a confirmation alert before issuing the refund**
Done by showing a simple `AlertViewController` in `RefundConfirmationViewController `

**Submit the refund to the API when the user confirms it.**
Updates `RefundConfirmationViewModel` to dispatch the `RefundStore.createRefund` action.

**Show a progress view while the refund is being created.**
Present an overlay controller from `RefundConfirmationViewController` powered by `InProgressViewController`

**Navigate back to the order details if the refund creation is successful.**
Dismiss the overlay and the modal refund flow after in `RefundConfirmationViewController` after the refund creation finishes

**Show an error notice if the refund creation fails**
If there is an error, show a generic notice and `DDLogError` the actual error.

**Update the order details screen to shows the new refund information.**
Update `RefundsStore` to update the refund's order - condensed refunds array to include the newly created refund.

# Gif
Success | Error
---- | ----
Refund flow! | error

# Testing Steps
**Success**
- Navigate to a non-refunded order
- Tap on the "Issue Refund" button
- Select some items to refund and tap next
- Tap on the "Refund" button
- See the order detail updated with the new refund information.

**Error**
- Navigate to a non-refunded order
- Tap on the "Issue Refund" button
- Select some items to refund and tap next
- Turn off any internet connection
- Tap on the "Refund" button
- See the error notice.

Update release notes:
- [X] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/Classes/ViewRelated/Orders/Order Details/Issue Refunds/RefundConfirmationViewModel.swift [hunk 1]
```diff
@@ -44,18 +48,34 @@ final class RefundConfirmationViewModel {
         )
     ]
 
-    init(details: Details, currencySettings: CurrencySettings = ServiceLocator.currencySettings) {
+    init(details: Details, actionProcessor: StoresManager = ServiceLocator.stores, currencySettings: CurrencySettings = ServiceLocator.currencySettings) {
```
Comment (ctarda) [inline WooCommerce/Classes/ViewRelated/Orders/Order Details/Issue Refunds/RefundConfirmationViewModel.swift:51]: I like this idiom. We get the best of both worlds. Having global dependencies clearly defined by a global entity and being able to inject dependencies without factories and local DI via the constructor without adding the rigidity that DI tends to add to all designs. Nice.

**Review by ctarda — APPROVED** [review]
I tried to break the flow by going back and forth, assigning items to be refunded, and then updating quantities or changing items and I couldn't 😄 

With an active connection:
1.  Alert shown refund:


2. After the refund is completed, the UI is updated and the refund is reflected


3. There is an error issuing the refund:


:shipit:

File: WooCommerce/Classes/ViewRelated/Orders/Order Details/Issue Refunds/RefundConfirmationViewModel.swift (hunk 1 above)
Comment (ctarda) [reply WooCommerce/Classes/ViewRelated/Orders/Order Details/Issue Refunds/RefundConfirmationViewModel.swift:51]: (Sorry, that was just me thinking out loud 😊)

Comment (Ecarrion) [reply WooCommerce/Classes/ViewRelated/Orders/Order Details/Issue Refunds/RefundConfirmationViewModel.swift:51]: 😄

**Review by Garance91540 — COMMENTED** [review]
Hi @Ecarrion,

Great work on refunds! Here are the few notes I made:

- **Disabled CTA styling should be updated.** 
I know I've seen a PR on that somewhere, so I know we are working on this. But on the Refund screen, the "Next" CTA should be in disabled state until at list one product has been selected, and right now it seems like we are showing a [pink empty CTA](https://d.pr/i/QSueFG). Here's how they should look in [light](https://d.pr/i/p0Prrz) and [dark](https://d.pr/i/AG1PiU) modes. (If you ever have a doubt on our general components, they can be found in the Figma file "System - iOS Components"

- **Update dialog CTAs styling**
I know you were working from outdated designs so that's on me, but in order to stay align with the styling we use for modals in other part of the app, the "Cancel" and "Refund" CTA  color should be iOS Sytem label / Light / Primary label color https://d.pr/i/vKm0vB

- Success snackbar
If possible let's add the success snackbar too when users successfully refunded products: https://d.pr/i/SEZ05U

**Comment by Ecarrion** [comment]
Hi @Garance91540 Thanks a lot for the review.

> Disabled CTA styling should be updated.

[An external contributor is working on this](https://github.com/woocommerce/woocommerce-ios/issues/3081). But I'll take care of it if it's not finished by this week.

> Update dialog CTAs styling

Updated
Screen Shot 2020-11-18 at 12 38 53 PM

> Success snackbar

Updated (Took the screenshot with a typo, but I have fixed it in code 😇 )
Screen Shot 2020-11-18 at 12 49 09 PM

**Review by Garance91540 — APPROVED** [review]
👏👏👏

**Pull request merged**